Doc/tutorial/datastructures.rst

```diff
@@ -48,10 +48,12 @@ objects:
    :noindex:
 
    Remove the item at the given position in the list, and return it.  If no index
-   is specified, ``a.pop()`` removes and returns the last item in the list.  (The
-   square brackets around the *i* in the method signature denote that the parameter
-   is optional, not that you should type square brackets at that position.  You
-   will see this notation frequently in the Python Library Reference.)
+   is specified, ``a.pop()`` removes and returns the last item in the list.
+   It raises an :exc:`IndexError` if the list is empty or the index is
+   outside the list range.  (The square brackets around the *i* in the method
```

Comment:
I also think that we can remove this note. Since the `[arg]` notation is used all over the docs, here is not the right place for it.

Reply:
I agree--didn't want to do that on my own accord, but yes.

Reply:
If you think something should be removed, you can ask in a comment.

Reply:
Sorry to comment after merge!

Re:

> (The square brackets around the i in the method signature denote that the parameter is optional, not that you should type square brackets at that position. You will see this notation frequently in the Python Library Reference.)

It's in the tutorial, which is introducing people to Python, and at the top of the page where this notation first appears.

[Diátaxis](https://diataxis.fr/tutorials/) says of tutorials:

> _A tutorial is an **experience** that takes place under the guidance of a tutor. A tutorial is always **learning-oriented**._

People do read through the tutorial from start to finish, and I think this is a good place to introduce and teach this notation.